### DISCUSSION TITLE
List with data shaping in REST API

### DISCUSSION THREAD

**Discussion by leomp12** [payloadcms/payload#2230] [Feature Requests & Ideas]
Would you be interested in a `fields` (or something) URL parameter to shape REST paginated results?
`GET /api/{collection-slug}?fields=title,link` for example

**Comment by DanRibbens** [comment]
We've avoided this as a feature so far and it's because of the overlap with GraphQL.

Querying nested structures using GraphQL is 100% more preferred than blasting query params into a REST GET. Also it would be a significant amount of code to reinvent the wheel.

I would be interested if anyone has a really compelling argument for why I'm wrong.

**Reply by leomp12** [reply]
IMO sometimes GraphQL is overhead.
Harder and more expensive to cache on proxy and specially CDNs.
And much harder to provide as API for third-party integrations, REST is much more well-know when you go out of frontend.

For example, I have a collection of orders with a very complex schema, ERPs must periodically consume these orders in the list, listing the complete orders will not be nice, a lot of bandwidth wasted, in the list they only need the status and a few more fields. But here we are talking about legacy software with proprietary language, it is not very easy to convince the integrator to learn GraphQL and implement a client in the legacy language of enterprise software 😄 

> Also it would be a significant amount of code to reinvent the wheel.

Anyway, I understand your position, that's why I asked first if it seemed interesting to you.

**Reply by DanRibbens** [reply]
These are all completely valid points about REST vs GraphQL.

Have you considered using access control to manage what data is being returned?

You could even write a function that used query params off the `req` if you wanted.
